mrmustard/physics/representations.py

```diff
@@ -0,0 +1,305 @@
+# Copyright 2023 Xanadu Quantum Technologies Inc.
+
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+
+#     http://www.apache.org/licenses/LICENSE-2.0
+
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+from __future__ import annotations
+
+import numpy as np
+from matplotlib import colors
+import matplotlib.pyplot as plt
+from mrmustard import math
+from mrmustard.physics import bargmann
+from mrmustard.physics.ansatze import Ansatz, PolyExpAnsatz
+from mrmustard.utils.typing import Batch, ComplexMatrix, ComplexTensor, ComplexVector, Scalar
+
+
+class Representation:
+    r"""
+    A base class for representations.
+    """
+    _contract_idxs: tuple[int, ...] = ()
```

Comment:
Small question about this protected attribute, or should it be an attribute of Class? It seems to be more like the Wires related and the Representation class only takes the responsibility to contract thing with given contract idxs.